ml-agents/mlagents/training_analytics_side_channel.py

```diff
@@ -0,0 +1,34 @@
+import uuid
+from mlagents_envs.exception import UnityCommunicationException
+from mlagents_envs.side_channel import SideChannel, IncomingMessage, OutgoingMessage
+
+from mlagents.trainers.settings import TrainerSettings
+
+
+class TrainingAnalyticsSideChannel(SideChannel):
+    """
+    Side channel that sends information about the training to the Unity environment so it can be logged.
+    """
+
+    def __init__(self) -> None:
+        # >>> uuid.uuid5(uuid.NAMESPACE_URL, "com.unity.ml-agents/TrainingAnalyticsSideChannel")
+        # UUID('b664a4a9-d86f-5a5f-95cb-e8353a7e8356')
+        super().__init__(uuid.UUID("b664a4a9-d86f-5a5f-95cb-e8353a7e8356"))
+
+    def on_message_received(self, msg: IncomingMessage) -> None:
+        raise UnityCommunicationException(
+            "The TrainingAnalyticsSideChannel received a message from Unity, "
+            + "this should not have happend."
+        )
+
+    def environment_initialized(self) -> None:
+        env_init_msg = OutgoingMessage()
```

Comment:
I'd like to encapsulate building all the data that gets sent from, instead of relying on it being passed in.

Reply:
How about creating an Event NamedTuple that gets created from the trainer settings? The data needs to come from somewhere, maybe from a trainer that is passed as argument. I think once the arguments are figured out, the name of the method should follow. Maybe something like "send_training_initialized_analytics_event" ?

Reply:
Sorry, I should clarify what I meant.

For environment_initialized(), I think we can grab most of what we want (python version, torch version) from global information, and don't need to pass it here as arguments. In this function, we'll make a proto message and serialize that as the payload.

For training_started(), we'll just pass in the TrainerSettings (like is currently happening) and make a different proto message here.